src/ray/gcs/gcs_server/test/gcs_placement_group_manager_test.cc

```diff
@@ -452,24 +512,20 @@ TEST_F(GcsPlacementGroupManagerTest, TestRescheduleWhenNodeDead) {
   ASSERT_EQ(placement_group->GetState(), rpc::PlacementGroupTableData::RESCHEDULING);
 
   // Test placement group rescheduling success.
+  mock_placement_group_scheduler_->placement_groups_.pop_back();
   OnPlacementGroupCreationSuccess(placement_group);
   ASSERT_EQ(placement_group->GetState(), rpc::PlacementGroupTableData::CREATED);
-  WaitForExpectedPgCount(1);
```

Comment:
This part of test makes no sense and uses APIs incorrectly. I moved this part to another test